pkg/admin/data/data_manager.go

```diff
@@ -224,48 +221,13 @@ func checkDirectoryContents(existing sets.Set[string]) error {
 }
 
 func copyPath(src, dest string) error {
-	tmpDest := fmt.Sprintf("%s.tmp", dest)
-	if exists, err := pathExists(tmpDest); err != nil {
+	copier := util.AtomicCopier{Source: src, Destination: dest}
+	if err := copier.CopyToIntermediate(); err != nil {
```

Comment:
What's the reason for exposing the intermediate / final implementation of the object? 
Should we just have a public `Copy` function and hide the implementation specifics?

Reply:
Do you mean the "ToIntermediate" part of the name?